backend/src/main/java/com/onlinebookstore/service/BookServiceImpl.java

```diff
@@ -10,14 +13,22 @@
 @RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Override
-    public Book save(Book book) {
+    public Book save(CreateBookRequest requestDto) {
```

Comment:
For the moment please return previous contract and move mapping to controller level
```suggestion
    public Book save(Book book) {
```